File: 프로그래머스/2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    answer = []
    period=[]
    for i in range(len(progresses)):
        day=1
        while (progresses[i]+speeds[i]<100):
            day+=1
            progresses[i]+=speeds[i]
        period.append(day)
    
    i=0
    k=1
    while i<len(period):
        cnt=1
        while (i+k<len(period) and period[i]>=period[i+k]):
            cnt+=1
            k+=1
        answer.append(cnt)
        i+=k
        k=1
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    let answer = [];
    let period = [];
    
    for (let i = 0; i < progresses.length; i++) {
        let day = 1;
        while (progresses[i] + speeds[i] < 100) {
            day += 1;
            progresses[i] += speeds[i];
        }
        period.push(day)
    }

    let day = period[0]
    let count = 1;
    for (let i = 1; i < period.length; i++) {
        if (day >= period[i]) {
            count++;
        } else {
            answer.push(count);
            day = period[i];
            count = 1;
        }        
    }
    answer.push(count);
    return answer;
}